File: Queue-Stack/Python/number_of_recent_calls.py
class RecentCounter:

    def __init__(self):
        self.maxx = []
        

    def ping(self, t: int) -> int:
        curmin, curmax = t-3000, t
        if not self.maxx:
            self.maxx.append(curmax)
            return 1
        self.maxx.append(curmax)
        while curmin > self.maxx[0]:
            self.maxx.pop(0)
        return len(self.maxx)
        
        


# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)

File: Queue-Stack/CPP/number_of_recent_calls.cpp
#include <queue>
using namespace std;

class RecentCounter {
public:
    queue<int>q;
    RecentCounter() {
    }
    
    int ping(int t) {
        int curmin = t - 3000;
        q.push(t);
        while (q.front() < curmin) {
            q.pop();
        }
        return q.size();
    }
};

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter* obj = new RecentCounter();
 * int param_1 = obj->ping(t);
 */